package NAlice.NWonderlogs;

import "alice/wonderlogs/protos/asr_prepared.proto";
import "alice/wonderlogs/protos/megamind_prepared.proto";
import "alice/wonderlogs/protos/uniproxy_prepared.proto";
import "alice/wonderlogs/protos/wonderlogs.proto";

message TErrorThresholdConfig {
    message TThreshold {
        oneof Value {
            uint64 Absolute = 1;
            double Share = 2;
        }
    }

    message TThresholds {
        optional TThreshold Warn = 1;
        optional TThreshold Crit = 2;
    }

    message TUniproxyPrepared {
        optional NWonderlogs.TUniproxyPrepared.TError.EProcess Process = 1;
        optional NWonderlogs.TUniproxyPrepared.TError.EReason Reason = 2;
        optional TThresholds Thresholds = 3;
    }

    message TMegamindPrepared {
        optional NWonderlogs.TMegamindPrepared.TError.EProcess Process = 1;
        optional NWonderlogs.TMegamindPrepared.TError.EReason Reason = 2;
        optional TThresholds Thresholds = 3;
    }

    message TAsrPrepared {
        optional NWonderlogs.TAsrPrepared.TError.EProcess Process = 1;
        optional NWonderlogs.TAsrPrepared.TError.EReason Reason = 2;
        optional TThresholds Thresholds = 3;
    }

    message TWonderlogs {
        optional TWonderlog.TError.EProcess Process = 1;
        optional TWonderlog.TError.EReason Reason = 2;
        optional TThresholds Thresholds = 3;
    }

    repeated TUniproxyPrepared UniproxyPrepared = 2;
    repeated TMegamindPrepared MegamindPrepared = 3;
    repeated TAsrPrepared AsrPrepared = 4;
    repeated TWonderlogs Wonderlogs = 5;
}
